lib/Frontend/Frontend.cpp

```diff
@@ -317,6 +317,23 @@ void CompilerInstance::setUpDiagnosticOptions() {
   }
 }
 
+// The ordering of ModuleLoaders is important!
+//
+// 1. SourceLoader: This is a hack and only the compiler's tests are using it,
+//    to avoid writing repetitive code involving generating modules/interfaces.
+//    Ideally, we'd get rid of it.
+// 2. MemoryBufferSerializedModuleLoader: This is used by LLDB, because it might
+//    already have the module available in memory.
+// 3. ModuleInterfaceLoader: Tries to find an up-to-date swiftmodule. If it
+//    succeeds, it issues a particular "error" (see
+//    [Note: ModuleInterfaceLoader-defer-to-SerializedModuleLoader]), which
+//    is interpreted by the overarching loader as a command to use the
+//    SerializedModuleLoader. If we failed to find a .swiftmodule, this falls
+//    back to using an interface. Actual errors lead to diagnostics.
+// 4. SerializedModuleLoader: Tries to load the serialized module.
```

Comment:
Nitpick: the "the" here is a little funny. What makes it "the" serialized module?

Reply:
In `ASTContext::getModule`, we load one module at a time using `loadModule`, so I figured it should be "the" serialized module. How would you phrase it -- "a serialized module"?

Reply:
That would work. The phrasing "the serialized module" assumes that every module we load has a serialized module, which isn't true. (The previous item also sorta falls prey to this.)

Reply:
I've changed it to "Loads a serialized module if it can".